src/modules/components/member/memberAvatar/memberAvatar.tsx

```diff
@@ -0,0 +1,63 @@
+import * as blockies from 'blockies-ts';
+import classNames from 'classnames';
+import type React from 'react';
+import { isAddress, type Hash } from 'viem';
+import { useEnsAddress, useEnsAvatar, useEnsName } from 'wagmi';
+import { Avatar, Spinner, type IAvatarProps } from '../../../../core';
+
+export interface IMemberAvatarProps extends Omit<IAvatarProps, 'fallback'> {
+    /**
+     * ENS name of the user to lookup avatar src.
+     */
+    ensName?: string;
+    /**
+     * 0x address of the user to look up ENS name and avatar src.
+     */
+    address?: string;
```

Comment:
Why not merge these two?

Reply:
> Why not merge these two?

I had it this way before as `ensNameOrAddress` -- it would depend on what comes back from the user object in the new backend in part, but separated does make less more readable validation logic.